frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/create-sc/create-sc.tsx

```diff
@@ -0,0 +1,263 @@
+    })
+    .catch(() => {
+      const requestData = getDiscoveryRequestData(state);
+      return k8sCreate(LocalVolumeDiscovery, requestData);
+    })
+    .then(() => {
+      dispatch({ type: 'setNodeNamesForLVS', value: selectedNodes });
+      onNext();
+      dispatch({ type: 'setIsLoading', value: false });
+    })
+    .catch((err) => {
+      dispatch({ type: 'setError', value: err.message });
+      dispatch({ type: 'setIsLoading', value: false });
+    });
+};
+
+const makeLocalVolumeSetCall = (state: State, dispatch: React.Dispatch<Action>) => {
+  dispatch({ type: 'setIsLoading', value: true });
+  const requestData = getLocalVolumeSetRequestData(state);
+  k8sCreate(LocalVolumeSetModel, requestData)
+    .then(() => {
+      state.onNextClick();
+      dispatch({ type: 'setIsLoading', value: false });
+    })
+    .catch((err) => {
+      dispatch({ type: 'setError', value: err.message });
+      dispatch({ type: 'setIsLoading', value: false });
+    });
+};
+
+const CreateSC: React.FC<CreateSCProps> = ({ match }) => {
+  const [state, dispatch] = React.useReducer(reducer, initialState);
+  const [nodeData, nodeLoaded, nodeLoadError] = useK8sWatchResource<NodeKind[]>(nodeResource);
+  const [discoveriesData, discoveriesLoaded, discoveriesLoadError] = useK8sWatchResource<
+    K8sResourceKind[]
+  >(nodesDiscoveriesResource);
+
+  React.useEffect(() => {
+    if (discoveriesLoaded && !discoveriesLoadError && discoveriesData.length) {
+      const nodesDiscoveries: NodesDiscoveries = discoveriesData.reduce((res, discovery) => {
+        const name = discovery?.spec?.nodeName;
+        const selectedNodes = getNodes(
+          state.showNodesListOnADV,
+          state.allNodeNamesOnADV,
+          state.nodeNamesForLVS,
+        );
+        if (selectedNodes.includes(name)) {
+          const obj = {} as NodeDiscoveries;
+          obj.node = name;
+          obj.discoveries = discovery?.status?.discoveredDevices ?? [];
+          res.push(obj);
+        }
+        return res;
+      }, []);
+
+      dispatch({ type: 'setNodesDiscoveries', value: nodesDiscoveries });
+      const capacity = getTotalDeviceCapacity(nodesDiscoveries);
+      dispatch({ type: 'setChartTotalData', value: capacity?.value });
+      dispatch({ type: 'setChartDataUnit', unit: capacity?.unit });
+    }
+  }, [
+    discoveriesData,
+    discoveriesLoaded,
+    discoveriesLoadError,
+    state.nodeNamesForLVS,
+    state.showNodesListOnADV,
+    state.allNodeNamesOnADV,
+  ]);
+  // displaying node names for discovery step
+  React.useEffect(() => {
+    if ((nodeLoadError || nodeData.length === 0) && nodeLoaded) {
+      dispatch({ type: 'setAllNodeNamesOnADV', value: [] });
+    } else if (nodeLoaded) {
+      const names = nodeData.filter((node) => !hasTaints(node)).map((node) => getName(node));
+      dispatch({ type: 'setAllNodeNamesOnADV', value: names });
+    }
+  }, [nodeData, nodeLoaded, nodeLoadError]);
```

Comment:
Yes, we need as its setting `allNodeNamesOnADV` state after filtering out the tainted nodes and this state is being used at various places. Since I am using the same logic in `create-sc.tsx` and `auto-detect-volume.tsx` both places, I will move this logic to just here `auto-detect-volume-inner.tsx`from where it can be used at these both places. Thanks for pointing it out.